raft/fsm.hh

```diff
@@ -27,7 +27,7 @@ struct fsm_output {
     std::vector<std::pair<server_id, rpc_message>> messages;
     // Entries to apply.
     std::vector<log_entry_ptr> committed;
-    std::optional<applied_snapshot> snp;
+    std::vector<applied_snapshot> snps;
```

Comment:
We discussed that on the call. This is the worst solution from all that has need discussed. There is no need to store outdated snapshot. Only the last one is relevant.

Reply:
I don't understand. This snapshots have already beed created, we just need to drop the outdated snapshots.

Reply:
Discussed in slack, the problem was that `_persistence->store_snapshot_descriptor` could be called for outdated snapshots which would be unnecessary. Perhaps we could return snapshot_id to be removed from `fsm::apply_snapshot`, but then that id would need to be passed through `snapshot_reply` so that the old snapshot could be dropped in `server_impl::apply_snapshot` . This seems to be more complicated than the current approach.